### ISSUE TITLE
[Bug]: svg.fonttype = 'none' is ignored if patheffects are set

### ISSUE THREAD

**Issue by mpflanzer** [matplotlib/matplotlib#26704]
### Bug summary

This might not be a bug per se but I think a different behaviour would make more sense. Currently the `svg.fonttype` parameter of the SVG backend is ignored if any path effect is set. Though I would argue that it might still desirable to apply path effects to the axis etc and then choose a font which is matching those effects instead of rendering the text with exactly the same effects.

In my case I noticed it with the "xckd" style plotting which also selects a matching font. But due to the path effects taking precedence the SVG would never use the font making it impossible to find (search) text in the SVG when displayed on a website. 

### Code for reproduction

```python
import matplotlib
import matplotlib.pyplot as plt

with plt.xkcd():
    plt.rcParams['svg.fonttype'] = 'none'

    fig = plt.figure()
    fig, ax = plt.subplots()
    ax.annotate("Test", (0.5, 0.5))
    plt.savefig("test.svg")
```

### Actual outcome

Instead of a `text` tag in the SVG the text is rendered as `path`

### Expected outcome

The text represented by a `text` tag in the SVG.

### Additional information

The "problem" originates in `text.py` line 785 (https://github.com/matplotlib/matplotlib/blob/781c6a0e8ac9a42f4079918b91d4310675dc1549/lib/matplotlib/text.py#L785) where the SVG renderer is swapped against the path effects renderer.

The behaviour I would prefer could be achieved with the following patch but it feels rather hacky. Any suggestion how to achieve this in a better way would be welcome.

```
diff --git a/lib/matplotlib/text.py b/lib/matplotlib/text.py
index 7a58ce7172..8197910820 100644
--- a/lib/matplotlib/text.py
+++ b/lib/matplotlib/text.py
@@ -782,7 +782,9 @@ class Text(Artist):
                     y = canvash - y
                 clean_line, ismath = self._preprocess_math(line)
 
-                if self.get_path_effects():
+                from matplotlib.backends.backend_svg import RendererSVG
+
+                if not (isinstance(renderer._renderer, RendererSVG) and mpl.rcParams['svg.fonttype'] == "none") and self.get_path_effects():
                     from matplotlib.patheffects import PathEffectRenderer
                     textrenderer = PathEffectRenderer(
                         self.get_path_effects(), renderer)
```

### Matplotlib Version

3.7.2

### Installation

pip

**Comment by ksunden** [comment]
I do not think we can ignore path effects for text generically, though there may be some latitude in the specific case of xkcd, as it looks like that may not actually alter text paths in the first place... (It is certainly not super noticeable with default settings if it does)

Path effects are actually _often_ used to modify text, though, in fact if you look at the [path effects guide](https://matplotlib.org/stable/tutorials/advanced/patheffects_guide.html) almost all of the examples of using it _are_ for text.

So then, if you have path effects, while in some cases (such as strokes or shadows), there are likely better ways of passing the info on to SVG and keeping things as text, the generic case of path effects are not necessarily trivial to implement in SVG (xkcd actually being among the harder ones to do so with, as it is parameterized and random-seeded)

Ideally most to all path effects could be implemented in svg and keep text instances, but that is not how the current implementation works for path effects, path effects are applied to mpl paths, and thus when applied to text, those text instances _must_ be paths.


These are a bit longer term projects, though would be welcome additions, if they could work.


That leaves us with options:
  - ignore path effects for text (non-viable, as text is one of the main usecases)
  - current behavior (ignore svg font type = none when path effects are in play) something needs to take precedence, and I don't think it is wrong for path effects to take precedence here.
  - error if font_type is `none` and path effects are applied (which would break a fair amount of existing code)
      - we could downgrade to a warning, but even then it is warning on expected behavior in many instances

From a library perspective I think I currently lean towards status quo in the near term (though implementing path effects in svg and being able to get both, at least _most_ of the time would be better).

Now, for what you want in this particular case, there is a pretty easy workaround that is explicit and gives you what you probably want:

```python
with plt.xkcd():
    plt.rcParams['svg.fonttype'] = 'none'

    fig = plt.figure()
    fig, ax = plt.subplots()
    ax.annotate("Test", (0.5, 0.5), path_effects=[])
    plt.savefig("test_path_effects_none.svg")
```

adding `path_effects=[]` to the annotate call negates the path effects applied to the text.

This works well for annotate and many other text calls (e.g. titles/labels/etc), though it may be a little trickier for tick labels. I think it may be that you have to loop through tick labels after they are created and calling `tick.set_path_effects([])` prior to saving/drawing... its not _ideal_ but it will work.

**Comment by mpflanzer** [comment]
Thanks for the background information on path effects.

> - current behavior (ignore svg font type = none when path effects are in play) something needs to take precedence, and I don't think it is wrong for path effects to take precedence here.
> - error if font_type is none and path effects are applied (which would break a fair amount of existing code)
>    - we could downgrade to a warning, but even then it is warning on expected behavior in many instances

For me this was not at all the expected behavior. Since I manually changed the `svg.fonttype` parameter I expected to see some effect of it. Or potentially a warning that it is ignored since paths effects are already set.
I would have chosen precedence the other way round since path effects (at least sound) more generic than just applying changes to text. Whereas the `svg.fonttype` is just about text. So for me it made sense to be able to enable path effects but then change the text rendering back to font based.

Anyway, I can live with the workaround you proposed. That is definitely good enough for my use case. Feel free to close this issue.
